### PR TITLE
Linked List Test

### PR DESCRIPTION
**Pull request by DemPaul**

### CODE REVIEW

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> first;
+    private Node<T> last;
+
+    private int countOfNodes = 0;
+
+    private class Node<N> {
+        private N item;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/MyLinkedList.java]: @DemPaul we are not pushing the solution! Please remove it. The solution should be provided in the separate file and shared with @Sofasmile via slack

File: src/test/java/core/basesyntax/MyLinkedListTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+    @Test
+    public void test1Add() {
+        myLinkedList.add("First");
+        myLinkedList.add("Second");
+        myLinkedList.add("Third");
+        linkedList.add("First");
+        linkedList.add("Second");
+        linkedList.add("Third");
+        Assert.assertEquals(linkedList.toString(), myLinkedList.toString());
```
Comment (boroda4436) [inline src/test/java/core/basesyntax/MyLinkedListTest.java]: @DemPaul You have overridden toString method in your solution. This is a bad idea. You shouldn't add any implementation to the _MyLinkedList_. Other students should do this. So this Assert is not useful. I guess comparing with _LinkedList_ in tests is a bad idea in the whole. You should take in a mind how _LinkedList_ works and expect the same behavior from _MyLinkedList_. Please refactor all tests where you are using toString comparing

File: src/test/java/core/basesyntax/MyLinkedListTest.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    @Test
+    public void test2Add() {
```
Comment (boroda4436) [inline src/test/java/core/basesyntax/MyLinkedListTest.java]: @DemPaul this is not a meaningful test case name

File: src/test/java/core/basesyntax/MyLinkedListTest.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        String expectedThird = linkedList.remove(3);
+        Assert.assertEquals(expectedFirst, actualFirst);
+        Assert.assertEquals(expectedSecond, actualSecond);
+        Assert.assertEquals(expectedThird, actualThird);
+        Assert.assertEquals(linkedList.toString(), myLinkedList.toString());
+    }
+
+    @Test
+    public void test7Remove() {
+        myLinkedList.remove("First");
```
Comment (boroda4436) [inline src/test/java/core/basesyntax/MyLinkedListTest.java]: @DemPaul what will happen if you will run just this one test case? Do you have elements? Each test case should be atomic

**Comment by boroda4436** [comment]
@DemPaul please add task description to the README file. Thanks

File: README.md [hunk 5]
```diff
@@ -1 +1,2 @@
 # jv-homework-template
+<h3>Реализовать свой LinkedList исходя из интерфейса https://pastebin.com/wMsifZSt
```
Comment (bchupika) [inline README.md]: @DemPaul please remove the reference

File: src/test/java/core/basesyntax/MyLinkedListTest.java [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+        myLinkedList.add(FIRST_ITEM);
+        boolean actualEmpty = myLinkedList.isEmpty();
+        Assert.assertTrue(actualInitialEmpty);
+        Assert.assertFalse(actualEmpty);
+    }
+
+    @Test
+    public void testIsEmptyAfterRemove() {
+        myLinkedList.add(FIRST_ITEM);
+        boolean actualEmpty = myLinkedList.isEmpty();
```
Comment (bchupika) [inline src/test/java/core/basesyntax/MyLinkedListTest.java]: @DemPaul you don't need this variable. Use assert immediately

File: src/test/java/core/basesyntax/MyLinkedListTest.java [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+        int actualSize = myLinkedList.size();
+        myLinkedList.remove(1);
+        int actualSizeAfterRemove = myLinkedList.size();
+        Assert.assertEquals(3, actualSize);
+        Assert.assertEquals(2, actualSizeAfterRemove);
+    }
+
+    @Test
+    public void testIsEmpty() {
+        boolean actualInitialEmpty = myLinkedList.isEmpty();
```
Comment (bchupika) [inline src/test/java/core/basesyntax/MyLinkedListTest.java]:
```suggestion
        Assert.assertTrue(myLinkedList.isEmpty());
```

File: src/test/java/core/basesyntax/MyLinkedListTest.java [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+        Assert.assertEquals(3, actualSize);
+        Assert.assertEquals(3, actualSizeAfterSet);
+    }
+
+    @Test
+    public void testSizeAfterRemove() {
+        myLinkedList.add(FIRST_ITEM);
+        myLinkedList.add(SECOND_ITEM);
+        myLinkedList.add(THIRD_ITEM);
+        int actualSize = myLinkedList.size();
```
Comment (bchupika) [inline src/test/java/core/basesyntax/MyLinkedListTest.java]:
```suggestion
        Assert.assertEquals(3, myLinkedList.size());
```

File: src/test/java/core/basesyntax/MyLinkedListTest.java [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+    }
+
+    @Test
+    public void testSizeAfterSet() {
+        myLinkedList.add(FIRST_ITEM);
+        myLinkedList.add(SECOND_ITEM);
+        myLinkedList.add(THIRD_ITEM);
+        int actualSize = myLinkedList.size();
+        myLinkedList.set(NEW_ITEM, 1);
+        int actualSizeAfterSet = myLinkedList.size();
```
Comment (bchupika) [inline src/test/java/core/basesyntax/MyLinkedListTest.java]:
```suggestion
        Assert.assertEquals(3, myLinkedList.size());
```

File: src/test/java/core/basesyntax/MyLinkedListTest.java [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+        Assert.assertNull(actualFourth);
+    }
+
+    @Test
+    public void testSize() {
+        int actualInitialSize = myLinkedList.size();
+        myLinkedList.add(FIRST_ITEM);
+        myLinkedList.add(SECOND_ITEM);
+        myLinkedList.add(THIRD_ITEM);
+        int actualSize = myLinkedList.size();
```
Comment (boroda4436) [inline src/test/java/core/basesyntax/MyLinkedListTest.java]: you don't need this variable. Use assert immediately
```suggestion
        Assert.assertEquals(3, myLinkedList.size());
```

File: src/test/java/core/basesyntax/MyLinkedListTest.java [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+    }
+
+    @Test
+    public void testSizeAfterRemove() {
+        myLinkedList.add(FIRST_ITEM);
+        myLinkedList.add(SECOND_ITEM);
+        myLinkedList.add(THIRD_ITEM);
+        int actualSize = myLinkedList.size();
+        myLinkedList.remove(1);
+        int actualSizeAfterRemove = myLinkedList.size();
```
Comment (boroda4436) [inline src/test/java/core/basesyntax/MyLinkedListTest.java]:
```suggestion
        Assert.assertEquals(2,  myLinkedList.size());
```

File: src/test/java/core/basesyntax/MyLinkedListTest.java (hunk 8 above)
Comment (boroda4436) [inline src/test/java/core/basesyntax/MyLinkedListTest.java]:
```suggestion
       Assert.assertEquals(3, myLinkedList.size());
```

File: README.md [hunk 12]
```diff
@@ -1 +1,2 @@
 # jv-homework-template
+<h3>Реализовать свой LinkedList исходя из интерфейса
```
Comment (boroda4436) [inline README.md]: @DemPaul Исходя из какого интерфейса?
```suggestion
<h3>Дано класс MyLinkedList который имплементирует интерфейс MyLinkedListInterface. Реализовать в нём свой LinkedList 
```

**Comment by boroda4436** [comment]
@Sofasmile please provide a review and merge the PR

**Pull request merged**